### PR TITLE
Add arm64 binary build for M1 chip

### PR DESCRIPTION
**Pull request by roee-landesman**
The intention of this PR is to add an `arm64` build of spin CLI so that it can be downloaded natively on M1 macbooks. This was discussed on this thread: https://github.com/spinnaker/spinnaker/issues/6632#issuecomment-1082726391

### CODE REVIEW

File: .github/workflows/build.yml
Comment (kskewes-sf) [inline .github/workflows/build.yml:58]:
```suggestion
          GOARCH=arm64 GOOS=darwin go build -ldflags "${LDFLAGS}" -o dist/${{ github.ref_name }}/${{ steps.build_variables.outputs.VERSION_WITHOUT_BRANCH }}/darwin/arm64/spin .
```

We need to target `darwin` OS.
Unfortunately the lines are slightly different between the 3 different files.

File: .github/workflows/pr.yml
Comment (kskewes-sf) [inline .github/workflows/pr.yml:49]:
```suggestion
          GOARCH=arm64 GOOS=darwin go build -ldflags "${LDFLAGS}" -o dist/${{ steps.build_variables.outputs.VERSION }}/darwin/arm64/spin .
```

File: .github/workflows/release.yml
Comment (kskewes-sf) [inline .github/workflows/release.yml:65]:
```suggestion
          GOARCH=arm64 GOOS=darwin go build -ldflags "${LDFLAGS}" -o dist/${{ steps.release_info.outputs.RELEASE_VERSION }}/darwin/arm64/spin .
```

**Review by kskewes-sf — CHANGES_REQUESTED** [review]

**Review by kskewes-sf — APPROVED** [review]
np. LGTM. We'll need a maintainer to review/approve and let CI run for us.

**Comment by mattgogerly** [comment]
Hi! Looks like the build is failing, any chance you could take a look? @kskewes-sf @roee-landesman

**Comment by roee-landesman** [comment]
Unfortunately I believe this may be because we are running go `1.15` precompiled from upstream. See: https://github.com/golang/go/issues/44703#issuecomment-788537428. Is there an effort to upgrade the spin CLI to newer versions of Go?

**Comment by kskewes-sf** [comment]
Good find. Yes let's upgrade. 
Maybe to 1.17 as 1.18 is new so might not be widely used or available by users of `spin` and there's also no patch version yet.
```
$ GOARCH=arm64 GOOS=darwin go1.15.15 build .
# github.com/spinnaker/spin
/home/kskewes/sdk/go1.15.15/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/tmp/go-link-542414637/go.o: file not recognized: file format not recognized
collect2: error: ld returned 1 exit status
```

```
$ GOARCH=arm64 GOOS=darwin go1.16.15 build .
$ file spin
spin: Mach-O 64-bit arm64 executable, flags:<|DYLDLINK|PIE>
```

**Comment by kskewes-sf** [comment]
See #335 - once merged that will unblock this PR.

**Comment by mattgogerly** [comment]
@Mergifyio update

**Review by mattgogerly — APPROVED** [review]

**Comment by kskewes-sf** [comment]
Can see binary here: https://console.cloud.google.com/storage/browser/spinnaker-artifacts/spin/master/f227e58-202204040731/darwin/arm64

**Pull request merged**
